### PR TITLE
LPD-24093 Removes attributes from request

### PR DESCRIPTION
**Pull request by ealonso**
Motivation
-----
If we have two liferay-ui:icon-menu rendered in the same page, one in a fragment and another one in a widget, the one in a widget is taking the values for the first one. It is because fragments don't have namespace, since those are not rendered inside a widget.

[Link to the ticket](https://liferay.atlassian.net/browse/LPD-24093)

Proposed Solution
-----
Removes attributes from request after rendering the tag, in that way if we have another taglib rendered after it we are going to have the correct values.

### CODE REVIEW

**Review by markocikos — COMMENTED** [review]
@ealonso The code in isolation looks good, but I don't quite understand how it fixes the issue. It's probably my gaps in knowledge about how wiring through request attributes works. See questions inline.

File: util-taglib/src/com/liferay/taglib/ui/IconMenuTag.java [hunk 1]
```diff
@@ -458,6 +458,12 @@ protected int processEndTag() throws Exception {
 
 		httpServletRequest.removeAttribute(
 			"liferay-ui:icon-menu:showWhenSingleIcon");
+		httpServletRequest.removeAttribute(
+			"liferay-ui:icon-menu:triggerAriaLabel");
+		httpServletRequest.removeAttribute(
+			"liferay-ui:icon-menu:triggerCssClass");
+		httpServletRequest.removeAttribute("liferay-ui:icon-menu:triggerLabel");
+		httpServletRequest.removeAttribute("liferay-ui:icon-menu:triggerType");
```
[the remark this answers is not in the document]
Comment (markocikos) [reply util-taglib/src/com/liferay/taglib/ui/IconMenuTag.java:466]: So, when we are getting the attributes, we first check non-namespaced ones and then namepaced ones?  :grimacing: I guess to make it work in both portlets and fragments?

In any case, it seems that cleanup on end tag is a good pattern, even more widely than just this case.

Thanks for clarifications!

**Review by markocikos — APPROVED** [review]
LGTM!

**Comment by ealonso** [comment]
ci:forward

File: util-taglib/src/com/liferay/taglib/ui/IconMenuTag.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (ealonso) [reply util-taglib/src/com/liferay/taglib/ui/IconMenuTag.java:466]: I think it is a question for Core Infra Team

**Pull request closed without merging**